sw/applications/example_power_gating_ram_blocks/example_power_gating_ram_blocks.c

```diff
@@ -0,0 +1,50 @@
+// Copyright EPFL contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdio.h>
+#include <stdlib.h>
+#include "csr.h"
+#include "hart.h"
+#include "handler.h"
+#include "core_v_mini_mcu.h"
+#include "power_manager.h"
+
+static power_manager_t power_manager;
+
+int main(int argc, char *argv[])
+{
+    // Setup power_manager
+    mmio_region_t power_manager_reg = mmio_region_from_addr(POWER_MANAGER_START_ADDRESS);
+    power_manager.base_addr = power_manager_reg;
+
+    power_manager_counters_t power_manager_ram_blocks_counters;
+
+    // Init ram block 2's counters
+    if (power_gate_counters_init(&power_manager_ram_blocks_counters, 40, 40, 30, 30) != kPowerManagerOk_e)
+    {
+        printf("Error: power manager fail. Check the reset and powergate counters value\n");
+        return EXIT_FAILURE;
+    }
+
+    // Power off ram block 2
+    if (power_gate_domain(&power_manager, kRam_2_e, kOff_e, &power_manager_ram_blocks_counters) != kPowerManagerOk_e)
+    {
+        printf("Error: power manager fail.\n");
+        return EXIT_FAILURE;
+    }
+
+    // Wait some time
+    for (int i=0; i<100; i++);
+
```

Comment:
Same comment (test one more interrupt here) and check the memories are indeed accessible again.

Reply:
Now, after wake-up, I write and read an array into the power-gated memory block. It should be enough to check the state.